### PR TITLE
journal: fix integer overflow in sd_journal_{next,previous}_skip()

### LINKED ISSUE #17502: possible integer overflow in journal/sd-journal.c: real_journal_next_skip
**systemd version the issue has been seen with**
 > git HEAD

**Used distribution**
 > N/A

**Linux kernel version used** (`uname -a`)
 > N/A

**CPU architecture issue was seen on**
 > N/A

**Expected behaviour you didn't see**
 > N/A

**Unexpected behaviour you saw**
 > integer overflow if API function sd_journal_next_skip is called with a skip parameter at value 9,223,372,036,854,775,808 or higher if the journal is large enough to support such . Line 808 will run into an integer overflow. The return value of your functions is defined to be positive on success and negative on error. With the integer overflow one may receive a negative value on a successful operation.

**Steps to reproduce the problem**
 > call function sd_journal_next_skip with a skip parameter at value 9,223,372,036,854,775,808 on a very large journal. I could not test it - my journal is too small and I am not bored enough to artificially blow up my journal to the required extend.

### PR DESCRIPTION
**Pull request by yuwata**
Fixes #17502.

### CODE REVIEW

**Comment by ente76** [comment]
Wouldn't it have been better to change the data type of skip? With this change things get messy for everyone using the library. Essentially I did limit skip - I doubt anyone needs a skip larger than 2^31.

File: src/journal/sd-journal.c [hunk 1]
```diff
@@ -879,7 +879,8 @@ _public_ int sd_journal_previous(sd_journal *j) {
 }
 
 static int real_journal_next_skip(sd_journal *j, direction_t direction, uint64_t skip) {
-        int c = 0, r;
+        uint64_t c = 0;
+        int r;
 
         assert_return(j, -EINVAL);
         assert_return(!journal_pid_changed(j), -ECHILD);
```
Comment (poettering) [inline src/journal/sd-journal.c:885]: Let's just refuse skips over INT_MAX here I'd say. It's cleaner and simpler

**Comment by yuwata** [comment]
@ente76 and @poettering Thank you for the comments. Updated. Now these functions refuse `skip` parameter larger than `INT_MAX`. PTAL.

**Comment by poettering** [comment]
lgtm

**Comment by poettering** [comment]
hmm, maybe ERANGE would be a better error to return?

**Comment by yuwata** [comment]
OK. Now they return `-ERANGE`.

**Pull request merged**